docker/Dockerfile

```diff
@@ -0,0 +1,102 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with IOTAManager.
+# If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by the GNU Affero General Public License
+# please contact with: [daniel.moranjimenez@telefonica.com]
+
+ARG NODE_VERSION=10.19.0-slim
+FROM node:${NODE_VERSION}
+ARG GITHUB_ACCOUNT=telefonicaid
+ARG GITHUB_REPOSITORY=iotagent-manager
+ARG DOWNLOAD=latest
+ARG SOURCE_BRANCH=master
+
+# Copying Build time arguments to environment variables so they are persisted at run time and can be 
+# inspected within a running container.
+# see: https://vsupalov.com/docker-build-time-env-values/  for a deeper explanation.
+
+ENV GITHUB_ACCOUNT=${GITHUB_ACCOUNT}
+ENV GITHUB_REPOSITORY=${GITHUB_REPOSITORY}
+ENV DOWNLOAD=${DOWNLOAD}
+
+MAINTAINER FIWARE IoTAgent Team. Telefónica I+D
+
+#
+# The following RUN command retrieves the source code from GitHub.
+# 
+# To obtain the latest stable release run this Docker file with the parameters
+# --no-cache --build-arg DOWNLOAD=stable
+# To obtain any speciifc version of a release run this Docker file with the parameters
+# --no-cache --build-arg DOWNLOAD=1.7.0
+#
+# The default download is the latest tip of the master of the named repository on GitHub
+#
+# Alternatively for local development, just copy this Dockerfile into file the root of the repository and 
+# replace the whole RUN statement by the following COPY statement in your local source using :
+#
+# COPY . /opt/iotaManager/
+#
+RUN apt-get update && \
+	# Ensure that unzip, wget and curl are installed
+	apt-get install -y --no-install-recommends unzip ca-certificates curl wget && \
+	if [ "${DOWNLOAD}" = "latest" ] ; \
+	then \
+		RELEASE="${SOURCE_BRANCH}"; \
+		echo "INFO: Building Latest Development from ${SOURCE_BRANCH} branch."; \
+	elif [ "${DOWNLOAD}" = "stable" ]; \
+	then \
+		RELEASE=$(curl -s https://api.github.com/repos/"${GITHUB_ACCOUNT}"/"${GITHUB_REPOSITORY}"/releases/latest | grep 'tag_name' | cut -d\" -f4); \
+		echo "INFO: Building Latest Stable Release: ${RELEASE}"; \
+	else \
+	 	RELEASE="${DOWNLOAD}"; \
+	 	echo "INFO: Building Release: ${RELEASE}"; \
+	fi && \
+	RELEASE_CONCAT=$(echo "${RELEASE}" | tr / -); \
+	wget --no-check-certificate -O source.zip https://github.com/"${GITHUB_ACCOUNT}"/"${GITHUB_REPOSITORY}"/archive/"${RELEASE}".zip && \
+	unzip source.zip && \
+	rm source.zip && \
+	mv "${GITHUB_REPOSITORY}-${RELEASE_CONCAT}" /opt/iotaManager && \
+	# Remove unzip and clean apt cache
+	apt-get clean && \
+	apt-get remove -y unzip && \
+	apt-get -y autoremove && \
+	rm -rf /var/lib/apt/lists/* 
+
+WORKDIR /opt/iotaManager
+
+RUN \
+	# Ensure that Git is installed prior to running npm install
+	apt-get update && \
+	apt-get install -y git && \
+	npm install pm2@3.2.2 -g && \
```

Comment:
It would be better to install pm2 only if the flag is enabled. I can be donde in the entrypoint

Reply:
Makes sense to me

Reply:
Done in 95622a78e7085d726f5590348f67095cdbecc02e but keep in mind that in case of PM2_ENABLED was true npm pm2 will be installed each time container starts.

Reply:
I'm ok with that. In any case it would be nice to test it, but there should not be any problem.